chain/chain/src/resharding.rs

```diff
@@ -419,6 +455,18 @@ impl Chain {
             &checked_account_id_to_shard_uid,
         )?;
 
+        #[cfg(feature = "yield_resume")]
```

Comment:
We may be able to work around that (do without runtime check) if we don't release yield resume with another resharding but this needs to be considered carefully. 

Technically we may even be able to do without it completely because this part of the resharding code works on blocks from the two epochs before the new protocol version so there may actually not be any overlap. 

In order to do this properly you'd need the protocol version of the correct block where resharding starts (where the snapshot is) and then throughout the epoch before the new protocol version when applying changes from parent shards to children shards. Each in the appropriate place, not all are here.

Reply:
On further thought, I don't think we gain anything from gating this within resharding. I've added [a small check](https://github.com/near/nearcore/pull/10833/files#diff-c0f06f89d72fe45495f45cff515bb3cf2a1bef45805d83403211048645f74548R340-R345) to make sure that this code performs a no-op if it gets run on trie state from an epoch without the feature enabled. Does that sound reasonable?

Reply:
Basically the checks should be elsewhere to make sure that promise yields don't make it into the trie state if they haven't been enabled (currently that is done with feature flags, I will fix that to check protocol version instead as you have pointed out).

Once the yields are in the trie state resharding may as well migrate them correctly; it doesn't need to concern itself directly with the protocol version.